impeller/entity/contents/text_contents.cc

```diff
@@ -28,22 +28,16 @@ void TextContents::SetTextFrame(TextFrame frame) {
   frame_ = std::move(frame);
 }
 
-void TextContents::SetGlyphAtlas(std::shared_ptr<GlyphAtlas> atlas) {
-  atlas_ = std::move(atlas);
-}
-
 void TextContents::SetGlyphAtlas(std::shared_ptr<LazyGlyphAtlas> atlas) {
-  atlas_ = std::move(atlas);
+  lazy_atlas_ = std::move(atlas);
 }
 
 std::shared_ptr<GlyphAtlas> TextContents::ResolveAtlas(
+    GlyphAtlas::Type type,
     std::shared_ptr<Context> context) const {
-  if (auto lazy_atlas = std::get_if<std::shared_ptr<LazyGlyphAtlas>>(&atlas_)) {
-    return lazy_atlas->get()->CreateOrGetGlyphAtlas(context);
-  }
-
-  if (auto atlas = std::get_if<std::shared_ptr<GlyphAtlas>>(&atlas_)) {
-    return *atlas;
+  FML_DCHECK(lazy_atlas_);
```

Comment:
Having the assertion as well as the conditional is redundant. I suggest removing the assert.

Reply:
The conditional is defensive - the assert is so that we don't get here in a test and return null, which would fail to render text.